flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -16,15 +16,200 @@
+                    Preconditions.checkNotNull(
+                            descriptor.findFieldByName(field.name().toLowerCase()));
+            // Get the value for a field.
+            // Check if the value is null.
+            @Nullable Object value = element.get(field.name());
+            if (value == null) {
+                // If the field required, throw an error.
+                if (fieldDescriptor.isRequired()) {
+                    throw new IllegalArgumentException(
+                            "Received null value for non-nullable field "
+                                    + fieldDescriptor.getName());
+                }
+            } else {
+                // Convert to Dynamic Message.
+                value = toProtoValue(fieldDescriptor, field.schema(), value);
+                builder.setField(fieldDescriptor, value);
+            }
+        }
+        return builder.build();
+    }
+
+    /**
+     * Function to convert a value of an AvroSchemaField value to required DynamicMessage value.
+     *
+     * @param fieldDescriptor {@link com.google.protobuf.Descriptors.FieldDescriptor} Object
+     *     describing the sink table field to which given value needs to be converted to.
+     * @param avroSchema {@link Schema} Object describing the value of Avro Schema Field.
+     * @param value Value of the Avro Schema Field.
+     * @return Converted Object.
+     * @throws UnsupportedOperationException In case schema is of type MAP.
+     */
+    private static Object toProtoValue(
+            FieldDescriptor fieldDescriptor, Schema avroSchema, Object value)
+            throws UnsupportedOperationException {
+        switch (avroSchema.getType()) {
+            case RECORD:
+                // Recursion
+                return getDynamicMessageFromGenericRecord(
+                        (GenericRecord) value, fieldDescriptor.getMessageType());
+            case ARRAY:
+                // Get an Iterable of all the values in the array.
+                Iterable<Object> iterable = (Iterable<Object>) value;
+                // Get the inner element type.
+                @Nullable Schema arrayElementType = avroSchema.getElementType();
+                if (arrayElementType.isNullable()) {
+                    throw new IllegalArgumentException("Array cannot have NULLABLE datatype");
+                }
+                if (arrayElementType.isUnion()) {
+                    throw new IllegalArgumentException(
+                            "ARRAY cannot have multiple datatypes in BigQuery.");
+                }
+                // Convert each value one by one.
+                return StreamSupport.stream(iterable.spliterator(), false)
+                        .map(v -> toProtoValue(fieldDescriptor, arrayElementType, v))
+                        .collect(Collectors.toList());
+            case UNION:
+                Schema type = BigQuerySchemaProviderImpl.handleUnionSchema(avroSchema).getLeft();
```

Comment:
Static methods like `handleUnionSchema` should not be imported from implicit implementations. Better to define in some common location and import from there.